src/scripts.js

```diff
@@ -251,45 +249,40 @@ function exitRecipe() {
 
 // TOGGLE DISPLAYS
 function showMyRecipesBanner() {
-  document.querySelector(".banner--message").style.display = "none";
-  document.querySelector(".banner--recipes").style.display = "block";
+  document.querySelector(".banner--message").style.display = "none"
+  document.querySelector(".banner--recipes").style.display = "block"
 }
 
 function showWelcomeBanner() {
-  document.querySelector(".banner--message").style.display = "block";
-  document.querySelector(".banner--recipes").style.display = "none";
+  document.querySelector(".banner--message").style.display = "block"
+  document.querySelector(".banner--recipes").style.display = "none"
 }
 
 // SEARCH RECIPES
+
 function pressEnterSearch(event) {
-  event.preventDefault();
-  searchRecipes();
+  event.preventDefault()
+  searchRecipes()
 }
 
 function searchRecipes() {
-  showAllRecipes();
-  let searchedRecipes = recipeData.filter(recipe => {
-    return recipe.name.toLowerCase().includes(searchInput.value.toLowerCase());
-  });
-  filterNonSearched(createRecipeObject(searchedRecipes));
-}
+  unhideRecipeCards();
+  const searchValue = searchInput.value.toLowerCase()
 
-function filterNonSearched(filtered) {
-  let found = recipeData.filter(recipe => {
-    let ids = filtered.map(f => f.id);
-    return !ids.includes(recipe.id)
+  const notSearchedNames = allRecipes.filter(recipe => {
+    return !recipe.name.toLowerCase().includes(searchValue)
+  })
+
+  const notSearchedRecipes = notSearchedNames.filter(recipe => {
+    return recipe.ingredients.some(ingredient => !ingredient.name.toLowerCase().includes(searchValue))
```

Comment:
Maybe return the second half of this line in brackets since it's a bit long but up to you. Otherwise good to go!